kpi/maintenance_tasks.py

```diff
@@ -0,0 +1,39 @@
+import constance
+from datetime import timedelta
+
+from django.db.models import Max
+from django.utils import timezone
+
+from kpi.models import AssetSnapshot
+
+
+default_retention_period = 90
+
+def remove_old_assetsnapshots():
+    days = (
+        constance.config.ASSET_SNAPSHOT_DAYS_RETENTION
+        or default_retention_period
+    )
+    max_snapshots_qs = (
+        AssetSnapshot.objects.exclude(asset_version=None)
```

Comment:
This could be a huge number. We need to chunk it. I get counts exceeding 4 million rows.

There was a distinct here that you removed, can you explain why it was unnecessary?

Why use the list?